swarm/pss/pss_test.go

```diff
@@ -61,431 +66,656 @@ func init() {
-		Action:  action,
-		Trigger: trigger,
-		Expect: &simulations.Expectation{
-			Nodes: recvids,
-			Check: check,
-		},
-	})
-	if result.Error != nil {
-		psslogmain.Error("msg failed!", "err", result.Error)
-		cancelmain()
-		t.Fatalf("simulation failed: %s", result.Error)
+	case cerr := <-lctx.Done():
+		t.Fatalf("test message timed out: %v", cerr)
 	}
-
-	wg.Wait()
-
-	if len(msgreceived) != msgcount {
-		t.Fatalf("Simulation Failed, got %d of %d msgs", len(msgreceived), msgcount)
-	}
-
-	psslogmain.Info("done!")
-	t.Logf("Simulation Passed, got %d of %d msgs", len(msgreceived), msgcount)
-	//t.Logf("Duration: %s", result.FinishedAt.Sub(result.StartedAt))
+	lsub.Unsubscribe()
+	rsub.Unsubscribe()
 }
 
-// triggerChecks triggers a simulation step check whenever a peer is added or
-// removed from the given node
-// connections and connectionstarget are temporary kademlia check workarounds
-func triggerChecks(ctx context.Context, wg *sync.WaitGroup, trigger *chan discover.NodeID, net *simulations.Network, id discover.NodeID) error {
+func TestCache(t *testing.T) {
+	var err error
+	var potaddr pot.Address
+	to, _ := hex.DecodeString("08090a0b0c0d0e0f1011121314150001020304050607161718191a1b1c1d1e1f")
+	//oaddr, _ := hex.DecodeString("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f")
+	//proofbytes, _ := hex.DecodeString("822fff7527f7ae630c1224921e50a7ca1b27324f00f3966623bd503780c7ab33")
+	keys, err := wapi.NewKeyPair()
+	privkey, err := w.GetPrivateKey(keys)
 
-	quitC := make(chan struct{})
+	ps := NewTestPss(privkey)
+	pp := NewPssParams(privkey)
+	data := []byte("foo")
+	datatwo := []byte("bar")
+	fwdaddr := network.RandomAddr()
+	copy(potaddr[:], fwdaddr.Over())
+	wparams := &whisper.MessageParams{
+		TTL:      DefaultTTL,
+		Src:      privkey,
+		Dst:      &privkey.PublicKey,
+		Topic:    PingTopic,
+		WorkTime: defaultWhisperWorkTime,
+		PoW:      defaultWhisperPoW,
+		Payload:  data,
+	}
+	woutmsg, err := whisper.NewSentMessage(wparams)
+	env, err := woutmsg.Wrap(wparams)
+	msg := &PssMsg{
+		Payload: env,
+		To:      to,
+	}
+	wparams.Payload = datatwo
+	woutmsg, err = whisper.NewSentMessage(wparams)
+	envtwo, err := woutmsg.Wrap(wparams)
+	msgtwo := &PssMsg{
+		Payload: envtwo,
+		To:      to,
+	}
 
-	node := net.GetNode(id)
-	if node == nil {
-		return fmt.Errorf("unknown node: %s", id)
+	digest, err := ps.storeMsg(msg)
+	if err != nil {
+		t.Fatalf("could not store cache msgone: %v", err)
 	}
-	client, err := node.Client()
+	digesttwo, err := ps.storeMsg(msgtwo)
 	if err != nil {
-		return err
+		t.Fatalf("could not store cache msgtwo: %v", err)
 	}
 
-	peerevents := make(chan *p2p.PeerEvent)
-	peersub, err := client.Subscribe(context.Background(), "admin", peerevents, "peerEvents")
-	if err != nil {
-		return fmt.Errorf("error getting peer events for node %v: %s", id, err)
+	// broken, don't know what to expect as whisper creates new msg every time
```

Comment:
yes it does but I'll remove the comment for now and remake one when the crypto stuff is approved